### PR TITLE
Add click-able local storage items

### LINKED ISSUE #7: Recent searches column: Using LocalStorage
https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API

cc @amrav @xtinct

### PR DESCRIPTION
**Pull request by icyflame**
- local storage is populated using the value of the query whenever the user clicked on a link
- every item in the local storage is a clickable item

The diff is slightly long because I have moved all the CSS and JS to separate files to make it easier to understand. Diffs for everything after the first commit should be easily understandable.

Earlier discussion was here: #12 

Live demo: http://icyflame.github.io/mfqp/

Screenshots:

image
image

Fixes #7

### CODE REVIEW

**Review by athityakumar — COMMENTED** [review]
Works. :+1: 

From a UX point of view, wouldn't it be better if the last searched query was on the top of the "Previous searches" list?

**Comment by icyflame** [comment]
@athityakumar Yeah, it would. Thanks for pointing it out! I will reverse chronological order is the one that makes sense.

**Comment by icyflame** [comment]
@vivekiitkgp @amrav Please review this PR. I will merge this in a few hours, after a little bit more testing, I feel confident about it though.

@athityakumar I have noted your suggestion, unfortunately, I can't work on them right now. I am adding an issue for now, and I will work on them later. I think we should push this out this exam session!

File: client.js
Comment (athityakumar) [inline client.js]: Line 32 of `client.js`

Won't `localStorage.setItem("searched",  query_value + "," + localStorage.getItem("searched"))` do the job of chronological order? 

Ping @icyflame

**Comment by athityakumar** [comment]
@icyflame - We should also probably add an issue for the mobile responsiveness, which we can add later. :smile:

File: client.js
Comment (icyflame) [reply client.js]: Only on the next reload of the page.

When the page loads, I get all the items from the LocalStorage and add an appropriate `<ul></ul>` with `<li>` elements. And then I simply append `<li>` elements to this list.

**Comment by icyflame** [comment]
@athityakumar I don't understand, do you think there's something wrong with the mobile view?

**Comment by athityakumar** [comment]
@icyflame - Nothing wrong, but it just doesn't look that awesome to me on mobile screen resolution.

image

File: client.js
Comment (athityakumar) [reply client.js]: Hmm, but the recent search does get appended to the list when a paper link is clicked right? Anyway, I think it'd be better if we can get started on the chronological order before merging this PR.

**Review by athityakumar — APPROVED** [review]
@icyflame - Thanks for making local storage work 🎉 . I think it's high time we merge this. What say?

**Comment by icyflame** [comment]
@athityakumar mobile preview still isn't good. that needs to be fixed. if you have time, can you get to it?

File: index.html
Comment (athityakumar) [inline index.html]: There's no local-storage.js file being used. I'll fix this along with UI part. Sending a PR in an hour.

Comment (icyflame) [reply index.html]: Great, rebase onto the latest master. Thanks! :+1:

**Comment by athityakumar** [comment]
@icyflame - Can you review https://github.com/icyflame/mfqp/pull/2 please? After that is merged, we can merge this too. 😄

**Comment by icyflame** [comment]
Hey @athityakumar, thanks for pointing me there. I will review and merge that. This is WAY over due! :smile:

**Comment by icyflame** [comment]
@amrav This patch is finally done! Can you please review this?

A live demo is available at http://icyflame.github.io/mfqp

## Preview on a computer

image

## Preview on a phone

screenshot_20171117-145921

**Comment by icyflame** [comment]
Hey @amrav, slight nudge for a review. :slightly_smiling_face:

**Comment by amrav** [comment]
@icyflame hey sorry for the delayed review! Code LGTM, but I'm not a fan of the two column layout. Google solves this by showing recent searches in the search bar itself, on focus, which is a neat solution. Another possibility is to have a two row layout, with a horizontal (wrapped) list of recent searches just beneath the search bar.

If you feel strongly about the current layout, I'm happy to merge as-is 👍

**Comment by thealphadollar** [comment]
@icyflame  PR has been stagnant for a long time.

**Comment by ghostwriternr** [comment]
I'm with Vikrant on this too. This two column layout seems a bit weird and pushes focus to the side. I like the simplicity that we have with MFQP right now. If something like the way Google does it can be added for these recent searches though, that would be pretty neat!

**Comment by icyflame** [comment]
@ghostwriternr @amrav noted. I have fixed the layout to show the recent searches after the results. This doesn't disturb the present UI. Since the discussion has gone on for long enough and this change is pretty small (UI wise), I will merge this in tomorrow unless someone raises some problems with the code. (I have re-indented the code so the diff might be a bit too big)

[Final Demo](https://www.siddharthkannan.in/mfqp/)

**Comment by icyflame** [comment]
Force pushed to rebase to latest gh-pages. The links weren't working before (old mfqp json file).

The diff isn't that big actually. It's easy to read too, so please do if you can.

Merging tomorrow morning.

**Pull request merged**
